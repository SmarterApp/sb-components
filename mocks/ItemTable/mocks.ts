import {
  ItemTableProps,
  ItemTableContainerProps,
  ItemCardModel
} from "@src/index";
import { aboutItemMockModel } from "@mocks/AboutItem/mocks";
import { itemCardList, sortableItemCards } from "@mocks/index";

export const tabClassNames = [
  "item",
  "claim",
  "standard",
  "subject",
  "grade",
  "item-type",
  "Target"
];

export const itemTableProps: ItemTableContainerProps = {
  isLinkTable: false,
  onRowSelection: () => {
    return;
  },
  onItemSelection: () => {
    return;
  },
  onCountNumberOfItemSelection: () => {
    return;
  },
  itemCards: itemCardList,
  item: {
    kind: "none"
  },

  numberOfSelectedItem: 0,
  getSelectedItemCount: () => {
    return 0;
  },
  showErrorModalOnPrintItemsCountExceeded: () => {
    return;
  },
  associatedItems: [],
  countNumberOfItemsAfterSelection: () => {
    return 0;
  }
};

export const itemTableSortProps: ItemTableContainerProps = {
  ...itemTableProps,
  item: {
    kind: "none"
  },
  itemCards: sortableItemCards
};

export const itemTableSelectProps: ItemTableContainerProps = {
  ...itemTableProps,
  item: {
    kind: "success",
    content: aboutItemMockModel
  }
};
