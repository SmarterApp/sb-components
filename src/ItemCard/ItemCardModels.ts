import { GradeLevels } from "../GradeLevels/GradeLevels";

export interface ItemCardModel {
  selected?: boolean;
  selectionIndex?: number;
  bankKey: number;
  itemKey: number;
  title: string;
  grade: GradeLevels;
  gradeLabel: string;
  subjectCode: string;
  subjectLabel: string;
  claimCode: string;
  claimLabel: string;
  targetHash: number;
  targetShortName: string;
  targetId: string;
  targetDescription: string;
  interactionTypeCode: string;
  interactionTypeLabel: string;
  isPerformanceItem: boolean;
  brailleOnlyItem?: boolean;
  domain?: string;
  depthOfKnowledge?: string;
  commonCoreStandardId?: string;
  ccssDescription?: string;
  calculator?: boolean;
}

export function itemIdEqual(a: ItemCardModel, b: ItemCardModel) {
  return a.itemKey === b.itemKey && a.bankKey === b.bankKey;
}
