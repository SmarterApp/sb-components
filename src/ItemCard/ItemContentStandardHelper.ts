export function getContentStandardCode(subjectCode:any, claimCode:any,  commonCoreStandardId:any, ccssDescription:any) {

    commonCoreStandardId = (commonCoreStandardId == null || commonCoreStandardId == undefined) ?  "Not Available" : commonCoreStandardId;
    ccssDescription = (ccssDescription == null || ccssDescription == undefined) ? "Content Standard information is currently unavailable for this item." : ccssDescription;
    
    //For math items and claim from 2 to 4
    if(subjectCode === 'MATH' && (claimCode == 'MATH2' || claimCode == 'MATH3' || claimCode == 'MATH4' )) {
      commonCoreStandardId = "Math Practice";
      ccssDescription = "Items in this claim primarily measure the Standards for Mathematical Practice rather than Content Standards.";
    }

    return {commonCoreStandardId, ccssDescription};
  }