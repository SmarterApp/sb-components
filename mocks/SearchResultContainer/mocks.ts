import {
  SearchResultContainerProps,
  SearchResultType,
  SearchAPIParamsModel
} from "@src/index";
import {
  itemCardList,
  totalItemsCardList,
  performanceTaskAssociatedItems
} from "@mocks/index";
import { mockSeachAPI } from "@mocks/ItemSearch/mocks";

export const mockSearchResultCardProps: SearchResultContainerProps = {
  isLinkTable: false,
  isInterimSite: true,
  totalItemCards: totalItemsCardList,
  showSelectAllButton: true,
  isPrintLimitEnabled: true,
  testItemsPool: [],
  onRowSelection: () => {
    return;
  },
  onItemSelection: () => {
    return;
  },

  onResetItems: () => {
    let visibleItemCardCopy = [];
    visibleItemCardCopy = itemCardList.slice();
    visibleItemCardCopy.forEach(element => {
      if (element.selected) {
        element.selected = false;
      }
    });
    return;
    //itemCardList = visibleItemCardCopy;
  },
  onSelectAll: () => {
    let visibleItemCardCopy = [];
    visibleItemCardCopy = itemCardList.slice();
    visibleItemCardCopy.forEach(element => {
      if (!element.selected) {
        element.selected = true;
      }
    });
    return;
  },
  onPrintItems: () => {
    return "";
  },
  searchAPI: mockSeachAPI,
  itemCards: itemCardList,
  item: {
    kind: "none"
  },
  defaultRenderType: SearchResultType.ItemCard,
  performanceTaskAssociatedItems: performanceTaskAssociatedItems
  // associatedItemsInPrintCart: [],
  // selectedItemsInPrintCart_props: []
};

export const mockSearchResultTableProps: SearchResultContainerProps = {
  ...mockSearchResultCardProps,
  defaultRenderType: SearchResultType.Table
};

export const mockSearchResultEmptyProps: SearchResultContainerProps = {
  ...mockSearchResultCardProps,
  itemCards: undefined,
  item: {
    kind: "none"
  },
  defaultRenderType: SearchResultType.ItemCard
};
