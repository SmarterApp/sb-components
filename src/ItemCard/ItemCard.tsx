import * as React from "react";
import * as GradeLevels from "../GradeLevels/GradeLevels";
import { ItemCardModel } from "./ItemCardModels";
import { Redirect } from "react-router";
import { ToolTip, generateTooltip } from "../index";
import { getContentStandardCode } from "./ItemContentStandardHelper";

// tslint:disable:no-require-imports
const claimIcons: { [claimCode: string]: string } = {
  MATH1: require("@sbac/sbac-ui-kit/src/images/math-1.svg"),
  MATH2: require("@sbac/sbac-ui-kit/src/images/math-2.svg"),
  MATH3: require("@sbac/sbac-ui-kit/src/images/math-3.svg"),
  MATH4: require("@sbac/sbac-ui-kit/src/images/math-4.svg"),
  ELA1: require("@sbac/sbac-ui-kit/src/images/ela-1.svg"),
  ELA2: require("@sbac/sbac-ui-kit/src/images/ela-2.svg"),
  ELA3: require("@sbac/sbac-ui-kit/src/images/ela-3.svg"),
  ELA4: require("@sbac/sbac-ui-kit/src/images/ela-4.svg")
};
// tslint:enable:no-require-imports

//props interface
export interface ItemCardProps {
  rowData: ItemCardModel;
  onRowSelect: (item: ItemCardModel) => void;
  getSelectedItemCount: () => number;
  showErrorModalOnPrintItemsCountExceeded: () => void;
  isPrintLimitEnabled: boolean;
  associatedItems: any[];
}

export interface ItemCardState {
  redirect: boolean;
  isCheckBoxChanged: boolean;
  isChecked: boolean;
  showErrorModal: boolean;
  statusMessage: string;
}

export class ItemCard extends React.Component<ItemCardProps, ItemCardState> {
  constructor(props: ItemCardProps) {
    super(props);
    this.state = {
      redirect: false,
      isChecked: false,
      isCheckBoxChanged: false,
      showErrorModal: false,
      statusMessage: ""
    };
  }

  shouldComponentUpdate(nextProps: ItemCardProps, nextState: ItemCardState) {
    // return nextState.redirect || nextState.isCheckBoxChanged;
    return true;
  }

  handleKeyPress = (e: React.KeyboardEvent<HTMLElement>) => {
    if (e.keyCode === 13 || e.keyCode === 23) {
      this.setState({ redirect: true });
    }
  };

  handleCheckBoxChange = (
    item: ItemCardModel,
    e: React.SyntheticEvent,
    shouldBeDisabled: string
  ) => {
    e.stopPropagation();
    e.nativeEvent.stopImmediatePropagation();
    if (
      shouldBeDisabled == "disabled" &&
      (item.selected === undefined || item.selected === false)
    ) {
      return;
    }
    const target = e.target as HTMLInputElement;
    const value = target.type === "checkbox" ? target.checked : target.value;
    let selectedItemsCount = this.props.getSelectedItemCount();
    //check if selection items count exceed the limits
    if (
      item.selected !== true &&
      this.props.isPrintLimitEnabled == true &&
      this.props.getSelectedItemCount() > 50
    ) {
      this.props.showErrorModalOnPrintItemsCountExceeded();
      return;
    } else {
      if (item.selected === true) item.selected = false;
      else item.selected = true;
      this.props.onRowSelect(item);
      //this.props.onItemSelection(item);
      this.setState({
        isChecked: value === true,
        isCheckBoxChanged: true
      });
    }
  };

  handleKeyUpEnterStopPropogation = (e: React.SyntheticEvent) => {
    e.stopPropagation();
  };

  handleEnterKeyDown = (e: React.KeyboardEvent) => {
    //if enter key is press prevent its default behaviour from selecting/clicking on elements
    if (e.keyCode === 13) {
      e.preventDefault();
    }
    /*>>>>>>> dev*/
  };

  handleOnClick = () => {
    this.setState({ redirect: true });
  };

  handleHideErrorModal = () => {
    this.setState({ showErrorModal: false, statusMessage: "" });
  };

  render() {
    /**
     * Function related to print button view
     */
    const onBtnClickChangeIcon = () => {
      return this.props.rowData.selected === true
        ? "fa-check-square"
        : "fa-plus-square";
    };
    const onBtnClickChangeBtnStyleCss = () => {
      return this.props.rowData.selected === true
        ? " btn-selected"
        : " btn-unselected";
    };
    const shouldBeDisabled = () => {
      if (
        this.props.rowData.selected === undefined ||
        this.props.rowData.selected === false
      ) {
        if (this.props.associatedItems !== undefined) {
          const associatedItems = this.props.associatedItems;
          const itemKey = this.props.rowData.itemKey;
          for (const itemKeyInAssociatedItems in associatedItems) {
            const associatedItemsArray =
              associatedItems[itemKeyInAssociatedItems];
            for (let i = 0; i < associatedItemsArray.length; i++) {
              console.log(associatedItemsArray[i][0].itemKey);
              if (associatedItemsArray[i][0].itemKey === itemKey)
                return "disabled";
            }
          }
        }
      }

      return " ";
    };
    const selectOrSelectedBtnText = () => {
      return this.props.rowData.selected === true
        ? " Item Selected"
        : " Select to Print";
    };

    const { bankKey, itemKey } = this.props.rowData;
    let content = <Redirect push to={`/Item/${bankKey}-${itemKey}`} />;

    if (!this.state.redirect) {
      const tooltip = generateTooltip({
        displayIcon: true,
        className: "box",
        helpText: <span>{this.props.rowData.targetDescription}</span>,
        displayText: this.props.rowData.targetId
      });

      const addOrRemoveIcon = () => {
        return this.props.rowData.selected === true ? "fa-minus" : "fa-plus";
      };
      const addOrRemoveIconClass = () => {
        return this.props.rowData.selected === true
          ? "btn-danger"
          : "btn-warning";
      };
      const getToolTipMsg = () => {
        if (addOrRemoveIcon() === "fa-plus") return "Add item to print bucket";
        else return "Remove item from print bucket ";
      };
      const iconsAddOrRemove = (
        <button
          className={"item-add-remove btn btn-sm btn-default"}
          onClick={e =>
            this.handleCheckBoxChange(this.props.rowData, e, shouldBeDisabled())
          }
          //onKeyUp={e => this.handleCheckboxKeyUpEnter(e, rowData)}
        >
          <i className={"fa fa-lg " + addOrRemoveIcon()} />
        </button>
      );
      const tooltip_addRemovePrintCart = generateTooltip({
        displayIcon: false,
        className: "",
        helpText: <span>{getToolTipMsg()}</span>,
        displayText: iconsAddOrRemove
      });
      // Tooltip for content standard
      const subjectCode = this.props.rowData.subjectCode;
      const claimCode = this.props.rowData.claimCode;
      let commonCoreStandardId = this.props.rowData.commonCoreStandardId;
      let ccssDescription = this.props.rowData.ccssDescription;
      //get the new and logically updated commonCoreStandardId, ccssDescription value
      const standard = getContentStandardCode(
        subjectCode,
        claimCode,
        commonCoreStandardId,
        ccssDescription
      );
      commonCoreStandardId = standard["commonCoreStandardId"];
      ccssDescription = standard["ccssDescription"];

      const tooltipCcontentStandard = generateTooltip({
        displayIcon: true,
        className: "box",
        helpText: <span>{ccssDescription}</span>,
        displayText: commonCoreStandardId
      });

      const tooltip_printOption = generateTooltip({
        displayIcon: true,
        className: "box",
        helpText: <span>Select to print this item</span>,
        displayText: ""
      });

      content = (
        <div
          role="link"
          className={`card card-block ${this.props.rowData.subjectCode.toLowerCase()}`}
          onClick={this.handleOnClick}
          onKeyUp={this.handleKeyPress}
          tabIndex={0}
        >
          <div className="card-contents">
            <div className="card-header">
              <h4 className="card-title">{this.props.rowData.subjectLabel}</h4>
              <div className="card-icon-container">
                {/* {tooltip_addRemovePrintCart} */}
                <span className="card-grade-tag card-icon">
                  {GradeLevels.GradeLevel.gradeCaseToShortString(
                    this.props.rowData.grade
                  )}
                </span>
              </div>
            </div>
            <p className="card-text grade">
              <span className="card-text-label">Grade:</span>
              <span className="card-text-value">
                {" "}
                {this.props.rowData.gradeLabel}
              </span>
            </p>
            <p className="card-text claim">
              <span className="card-text-label">Claim:</span>
              <span className="card-text-value">
                {" "}
                {this.props.rowData.claimLabel}
              </span>
            </p>
            <p className="card-text target">
              <span className="card-text-label">Target:</span>
              <span className="card-text-value">{tooltip}</span>
            </p>
            <p className="card-text target">
              <span className="card-text-label">Standard:</span>
              <span className="card-text-value">{tooltipCcontentStandard}</span>
            </p>
            <p className="card-text interaction-type">
              <span className="card-text-label">Item Type:</span>
              <span className="card-text-value">{` ${
                this.props.rowData.interactionTypeLabel
              }`}</span>
            </p>
            <p className="card-text item-id">
              <span className="card-text-label">Item Id:</span>
              <span className="card-text-value">
                {" "}
                {this.props.rowData.itemKey}
              </span>
            </p>
            <button
              type="button"
              className={`btn btn-add-remove-print-selection ${this.props.rowData.subjectCode.toLowerCase()} ${onBtnClickChangeBtnStyleCss()}`}
              onClick={e =>
                this.handleCheckBoxChange(
                  this.props.rowData,
                  e,
                  shouldBeDisabled()
                )
              }
              tabIndex={0}
              onKeyUp={e => this.handleKeyUpEnterStopPropogation(e)}
              onKeyDown={e => this.handleEnterKeyDown(e)}
            >
              <i className={"fa  " + onBtnClickChangeIcon()} />
              &nbsp;&nbsp;
              {selectOrSelectedBtnText()}
            </button>
          </div>
          {/* <button
            type="button"
            className={`btn btn-default btn-add-remove-print-selection ${this.props.rowData.subjectCode.toLowerCase()} ${onBtnClickChangeBtnStyleCss()}`}
            onClick={e => this.handleCheckBoxChange(this.props.rowData, e)}
          >
            <i className={"fa  " + onBtnClickChangeIcon()} />&nbsp;&nbsp;
            {selectOrSelectedBtnText()}
          </button> */}
        </div>
      );
    }

    return (
      <>
        {/* {this.renderErrorModal()} */}
        {content}
      </>
    );
  }
}
