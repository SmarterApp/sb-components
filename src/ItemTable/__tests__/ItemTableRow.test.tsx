import "jsdom-global/register";
import * as React from "react";
import { shallow, mount, render } from "enzyme";
import { itemCardList, testCodeToLabelMap_mock } from "@mocks/ItemCard/mocks";
import { itemHandler } from "./mocks";
import {
  GradeLevels,
  RubricModel,
  AboutItemModel,
  Resource,
  ItemCardModel,
  headerColumns,
  ItemTableRowProps,
  ItemTableRow
} from "@src/index";
import { getColumnsHeaderConfig_mock } from "../mocks";

describe("ItemTableRow", () => {
  const rubrics: RubricModel[] = [];
  const selectedItem = itemCardList[0];

  const props: ItemTableRowProps = {
    rowData: selectedItem,
    hasControls: true,
    columns: headerColumns,
    isExpanded: false,
    onRowExpand: itemHandler,
    onRowSelect: itemHandler,
    isInterimSite: true,
    testCodeToLabelMap: testCodeToLabelMap_mock,
    associatedItems: [],
    countNumberOfItemsAfterSelection: () => 1,
    isItemSelected: true,
    getSelectedItemCount: () => 4,
    itemTableConfig: getColumnsHeaderConfig_mock(),
    numberOfSelectedItem: 4,
    showErrorModalOnPrintItemsCountExceeded: () => false
  };

  const wrapper = mount(
    <table>
      <tbody>
        <ItemTableRow {...props} />
      </tbody>
    </table>
  );
  const wrapperExpanded = shallow(
    <ItemTableRow {...props} isExpanded={true} />
  );

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("matches expand snapshot", () => {
    expect(wrapperExpanded).toMatchSnapshot();
  });

  it("calls row events", () => {
    const item = wrapper.find("td.item");
    item.simulate("click");
    expect(props.onRowSelect).toHaveBeenCalled();
    expect(props.onRowExpand).toHaveBeenCalled();
  });
});
