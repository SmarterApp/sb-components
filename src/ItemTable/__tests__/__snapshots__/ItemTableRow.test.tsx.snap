// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTableRow matches expand snapshot 1`] = `
<tr
  className="selected"
  key="187-3206"
  onClick={[Function]}
  onKeyUp={[Function]}
>
  <React.Fragment>
    <ToolTip
      className=""
      displayIcon={false}
      displayText={
        <td
          className="item-checkbox item-add-remove"
          onClick={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
        >
          <button
            className="btn btn  btn-item-add-remove-grid btn-sm btn-blue-border"
            type="button"
          >
            <i
              className="fa fa-2x fa-plus-circle"
            />
          </button>
        </td>
      }
      helpText={
        <label>
          Add item to print queue
        </label>
      }
    />
  </React.Fragment>
  <td
    className="item"
    key="item-187-3206"
    role="gridcell"
  >
    <span
      key="item"
    >
      <a
        role="link"
        tabIndex={0}
      >
        3206
      </a>
    </span>
  </td>
  <td
    className="subject"
    key="subject-187-3206"
    role="gridcell"
  >
    <span
      key="subject"
    >
      <span
        className="table-subject-highlight math"
      >
        Math
      </span>
    </span>
  </td>
  <td
    className="grade"
    key="grade-187-3206"
    role="gridcell"
  >
    <span
      key="grade"
    >
      <span>
        Grade 6
      </span>
    </span>
  </td>
  <td
    className="claim"
    key="claim-187-3206"
    role="gridcell"
  >
    <span
      key="claim"
    >
      <span>
        Concepts and Procedures and Logical
      </span>
    </span>
  </td>
  <td
    className="Target"
    key="Target-187-3206"
    role="gridcell"
  >
    <span
      key="target"
    >
      <ToolTip
        className="box"
        displayIcon={true}
        displayText="A"
        helpText={
          <p>
            Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia
          </p>
        }
        position="bottom"
      />
    </span>
  </td>
  <td
    className="standard"
    key="standard-187-3206"
    role="gridcell"
  >
    <span
      key="standard"
    >
      <span>
        6.M.1a
      </span>
    </span>
    <span
      key="standard"
    >
      <ToolTip
        className="box"
        displayIcon={true}
        displayText=""
        helpText={
          <p>
            Content Standard information is currently unavailable for this item.
          </p>
        }
        position="bottom"
      />
    </span>
  </td>
  <td
    className="item-type"
    key="item-type-187-3206"
    role="gridcell"
  >
    <span
      key="item-type"
    >
      <span>
        Equation Multiple Choice choice
      </span>
    </span>
  </td>
</tr>
`;

exports[`ItemTableRow matches snapshot 1`] = `
<table>
  <tbody>
    <ItemTableRow
      columns={
        Array [
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "item",
              },
            ],
            "compare": [Function],
            "header": "Item",
            "headerClassName": "item",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "subject",
              },
            ],
            "compare": [Function],
            "header": "Subject",
            "headerClassName": "subject",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "grade",
              },
            ],
            "compare": [Function],
            "header": "Grade",
            "headerClassName": "grade",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "claim",
              },
            ],
            "compare": [Function],
            "header": "Claim",
            "headerClassName": "claim",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "target",
                "helpText": [Function],
              },
            ],
            "compare": [Function],
            "header": "Target",
            "headerClassName": "Target",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "standard",
              },
              Object {
                "accessor": [Function],
                "className": "standard",
                "helpText": [Function],
              },
            ],
            "compare": [Function],
            "header": "Standard",
            "headerClassName": "standard",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "item-type",
              },
            ],
            "compare": [Function],
            "header": "Item Type",
            "headerClassName": "item-type",
          },
        ]
      }
      hasControls={true}
      isExpanded={false}
      onRowExpand={[MockFunction]}
      onRowSelect={[MockFunction]}
      rowData={
        Object {
          "bankKey": 187,
          "brailleOnlyItem": false,
          "calculator": true,
          "claimCode": "MATH1",
          "claimLabel": "Concepts and Procedures and Logical",
          "commonCoreStandardId": "6.M.1a",
          "grade": 8,
          "gradeLabel": "Grade 6",
          "interactionTypeCode": "EQ",
          "interactionTypeLabel": "Equation Multiple Choice choice",
          "isPerformanceItem": true,
          "itemKey": 3206,
          "subjectCode": "MATH",
          "subjectLabel": "Math",
          "targetDescription": "Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia",
          "targetHash": 100,
          "targetId": "A",
          "targetShortName": "",
          "title": "Math Grade 6 Claim 1",
        }
      }
    >
      <tr
        className=""
        key="187-3206"
        onClick={[Function]}
        onKeyUp={[Function]}
      >
        <ToolTip
          className=""
          displayIcon={false}
          displayText={
            <td
              className="item-checkbox item-add-remove"
              onClick={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <button
                className="btn btn  btn-item-add-remove-grid btn-sm btn-blue-border"
                type="button"
              >
                <i
                  className="fa fa-2x fa-plus-circle"
                />
              </button>
            </td>
          }
          helpText={
            <label>
              Add item to print queue
            </label>
          }
        >
          <span
            className="tool-tip-links "
            tabIndex={0}
          >
            <span
              className="tool-tip-hoverable"
            >
              <td
                className="item-checkbox item-add-remove"
                onClick={[Function]}
                onKeyUp={[Function]}
                tabIndex={0}
              >
                <button
                  className="btn btn  btn-item-add-remove-grid btn-sm btn-blue-border"
                  type="button"
                >
                  <i
                    className="fa fa-2x fa-plus-circle"
                  />
                </button>
              </td>
               
            </span>
            <span
              className="tool-tip-details"
            >
              <span
                className="tool-tip-message  "
              >
                <label>
                  Add item to print queue
                </label>
              </span>
            </span>
          </span>
        </ToolTip>
        <td
          className="item"
          key="item-187-3206"
          role="gridcell"
        >
          <span
            key="item"
          >
            <a
              role="link"
              tabIndex={0}
            >
              3206
            </a>
          </span>
        </td>
        <td
          className="subject"
          key="subject-187-3206"
          role="gridcell"
        >
          <span
            key="subject"
          >
            <span
              className="table-subject-highlight math"
            >
              Math
            </span>
          </span>
        </td>
        <td
          className="grade"
          key="grade-187-3206"
          role="gridcell"
        >
          <span
            key="grade"
          >
            <span>
              Grade 6
            </span>
          </span>
        </td>
        <td
          className="claim"
          key="claim-187-3206"
          role="gridcell"
        >
          <span
            key="claim"
          >
            <span>
              Concepts and Procedures and Logical
            </span>
          </span>
        </td>
        <td
          className="Target"
          key="Target-187-3206"
          role="gridcell"
        >
          <span
            key="target"
          >
            <ToolTip
              className="box"
              displayIcon={true}
              displayText="A"
              helpText={
                <p>
                  Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia
                </p>
              }
              position="bottom"
            >
              <span
                className="tool-tip-links box"
                tabIndex={0}
              >
                <span
                  className="tool-tip-hoverable"
                >
                  A
                   
                  <span
                    aria-hidden="true"
                    className="fa fa-info-circle fa-sm"
                  />
                </span>
                <span
                  className="tool-tip-details"
                >
                  <span
                    className="tool-tip-message bottom "
                  >
                    <p>
                      Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia
                    </p>
                  </span>
                </span>
              </span>
            </ToolTip>
          </span>
        </td>
        <td
          className="standard"
          key="standard-187-3206"
          role="gridcell"
        >
          <span
            key="standard"
          >
            <span>
              6.M.1a
            </span>
          </span>
          <span
            key="standard"
          >
            <ToolTip
              className="box"
              displayIcon={true}
              displayText=""
              helpText={
                <p>
                  Content Standard information is currently unavailable for this item.
                </p>
              }
              position="bottom"
            >
              <span
                className="tool-tip-links box"
                tabIndex={0}
              >
                <span
                  className="tool-tip-hoverable"
                >
                   
                  <span
                    aria-hidden="true"
                    className="fa fa-info-circle fa-sm"
                  />
                </span>
                <span
                  className="tool-tip-details"
                >
                  <span
                    className="tool-tip-message bottom "
                  >
                    <p>
                      Content Standard information is currently unavailable for this item.
                    </p>
                  </span>
                </span>
              </span>
            </ToolTip>
          </span>
        </td>
        <td
          className="item-type"
          key="item-type-187-3206"
          role="gridcell"
        >
          <span
            key="item-type"
          >
            <span>
              Equation Multiple Choice choice
            </span>
          </span>
        </td>
      </tr>
    </ItemTableRow>
  </tbody>
</table>
`;
